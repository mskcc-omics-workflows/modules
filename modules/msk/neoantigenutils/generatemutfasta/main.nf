process NEOANTIGENUTILS_GENERATEMUTFASTA {
    tag "$meta.id"
    label 'process_single'
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/neoantigen-utils-base:1.0.0':
        'docker.io/mskcc/neoantigen-utils-base:1.0.0' }"

    input:
    tuple val(meta),  path(inputMaf)
    tuple path(cds),  path(cdna)

    output:
    tuple val(meta), path("*_out/*.MUT_sequences.fa"),     emit: mut_fasta
    tuple val(meta), path("*_out/*.WT_sequences.fa"),      emit: wt_fasta
    path "versions.yml",                                   emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    mkdir ${prefix}_out

    generateMutFasta.py --sample_id ${prefix} \
    --output_dir ${prefix}_out \
    --maf_file ${inputMaf} \
    --CDS_file ${cds} \
    --CDNA_file ${cdna}


    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        generateMutFasta: \$(echo \$(generateMutFasta.py -v))
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
        mkdir ${prefix}_out
        touch ${prefix}_out/${prefix}.MUT_sequences.fa
        touch ${prefix}_out/${prefix}.WT_sequences.fa
        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            generateMutFasta: \$(echo \$(generateMutFasta.py -v))
        END_VERSIONS
    """
}
