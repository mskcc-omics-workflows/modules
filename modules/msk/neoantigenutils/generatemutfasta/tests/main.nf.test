nextflow_process {

    name "Test Process NEOANTIGENUTILS_GENERATEMUTFASTA"
    script "../main.nf"
    process "NEOANTIGENUTILS_GENERATEMUTFASTA"

    tag "modules"
    tag "modules_nfcore"
    tag "neoantigenutils"
    tag "neoantigenutils/generatemutfasta"

    test("neoantigenutils_generatemutfasta - maf - fasta") {

        when {

            process {
                """

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['temp_test_maf'], checkIfExists: true)
                    ]

                input[1] = [
                    file(params.test_data_mskcc['neoantigen']['cds'], checkIfExists: true),
                    file(params.test_data_mskcc['neoantigen']['cdna'], checkIfExists: true)
                    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }


    test("neoantigenutils_generatemutfasta - maf - fasta - stub") {

        options "-stub"

        when {

            process {
                """

                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.test_data_mskcc['neoantigen']['temp_test_maf'], checkIfExists: true)
                    ]

                input[1] = [
                    file(params.test_data_mskcc['neoantigen']['cds'], checkIfExists: true),
                    file(params.test_data_mskcc['neoantigen']['cdna'], checkIfExists: true)
                    ]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }



        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
