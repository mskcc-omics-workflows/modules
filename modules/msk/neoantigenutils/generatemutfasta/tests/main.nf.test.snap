{
    "neoantigenutils_generatemutfasta - maf - fasta": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.MUT_sequences.fa:md5,c236ca28e7c658b74377c19437d7c5ab"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.WT_sequences.fa:md5,b1a6ea2978a6624c98112e8118658495"
                    ]
                ],
                "2": [
                    "versions.yml:md5,0fa44b299ab4ec385f2cc6831bbc6923"
                ],
                "mut_fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.MUT_sequences.fa:md5,c236ca28e7c658b74377c19437d7c5ab"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0fa44b299ab4ec385f2cc6831bbc6923"
                ],
                "wt_fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.WT_sequences.fa:md5,b1a6ea2978a6624c98112e8118658495"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-22T17:16:33.363826904"
    },
    "neoantigenutils_generatemutfasta - maf - fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.MUT_sequences.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.WT_sequences.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,0fa44b299ab4ec385f2cc6831bbc6923"
                ],
                "mut_fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.MUT_sequences.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0fa44b299ab4ec385f2cc6831bbc6923"
                ],
                "wt_fasta": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.WT_sequences.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2024-11-22T17:16:40.580416341"
    }
}